comps/dataprep/multimodal/redis/langchain/prepare_videodoc_redis.py

```diff
@@ -580,6 +591,52 @@ async def ingest_videos_with_transcripts(files: List[UploadFile] = File(None)):
 
             print(f"Processed video {video_file.filename}")
 
+        for image_file in image_files:
+            print(f"Processing image {image_file.filename}")
+
+            # Assign unique identifier to image
+            image_id = generate_id()
+
+            # Create image file name by appending identifier
+            image_name, image_ext = os.path.splitext(image_file.filename)
+            image_file_name = f"{image_name}_{image_id}{image_ext}"
+            image_dir_name = os.path.splitext(image_file_name)[0]
+
+            # Save image file in upload_directory
+            with open(os.path.join(upload_folder, image_file_name), "wb") as f:
+                shutil.copyfileobj(image_file.file, f)
+            uploaded_videos_saved_videos_map[image_name] = image_file_name
+
+            # Save captions file in upload directory
+            txt_file_name = os.path.splitext(image_file.filename)[0] + ".txt"
+            txt_idx = None
+            for idx, caption_file in enumerate(captions_files):
```

Comment:
Are you saying that iterating through the image files and then the whole list of caption files looking for the match for each one is inefficient? It would be better to match them up once and then iterate through the pairs? I could do that.

Reply:
Yeah, the lists are already being iterated above where there's a check to make sure that there is a captions file for every image, so at that point, you could make a captions list that matches the order of the files list. If you had that, I don't think you'd need this loop.